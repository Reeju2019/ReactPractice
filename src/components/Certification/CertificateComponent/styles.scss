.share-btn {
  font-size: 1.1rem;
  font-weight: 500;
  color: #6851b5;
}

.share-btn:hover {
  cursor: pointer;
}

// styles for template-1

.main-box-div-template-1 {
  position: relative;
  .img-bg-template-1.col {
    img {
      max-width: 100%;
      min-height: 200px;
    }
  }
  .content-template-1 {
    position: absolute;
    top: 0;
    left: 0;
    padding: 30px 70px;
  }
}

.template-1-header-1 {
  font-size: 0.9rem;
  opacity: 0.7;
  font-weight: 700;
}

.header-2-violet {
  font-size: 1rem;
  opacity: 0.8;
  font-weight: 500;
  color: #4326a3;
  margin: 0;
  line-height: 0%;
}

.template-1-header-3 {
  font-size: 0.8rem;
  font-weight: 500;
  opacity: 0.8;
  line-height: 0%;
}

.template-1-header-4 {
  font-size: 0.9rem;
  opacity: 0.9;
  font-weight: 500;
}

.template-1-header-5 {
  font-size: 0.8rem;
  opacity: 0.8;
}

@media screen and (max-width: 991px) {
  .main-box-div-template-1 {
    position: relative;
    .img-bg-template-1.col {
      img {
        max-width: 100%;
        height: 590px;
      }
    }
  }
}

@media screen and (max-width: 767px) {
  .main-box-div-template-1 {
    position: relative;
    .img-bg-template-1.col {
      img {
        max-width: 100%;
        height: 710px;
      }
    }
  }
}

@media screen and (max-width: 576px) {
  .main-box-div-template-1 {
    position: relative;
    .img-bg-template-1.col {
      img {
        max-width: 100%;
        height: 850px;
      }
    }
  }
}

// styles for template-02

.main-box-div-template-2 {
  position: relative;
  .img-bg-template-2.col {
    img {
      max-width: 100%;
      min-height: 200px;
    }
  }
  .content-template-2 {
    position: absolute;
    top: 0;
    left: 0;
    padding: 30px 70px;
  }
}

.template-2-header-1 {
  font-size: 1.5rem;
  font-weight: 500;
}

.template-2-header-2 {
  font-size: 1.2rem;
  opacity: 0.8;
  font-weight: 500;
}

.template-2-header-2-blue {
  font-size: 1.2rem;
  opacity: 0.9;
  font-weight: 500;
  color: #114eaf;
  margin: 0;
}

.template-2-header-3 {
  font-size: 1rem;
  font-weight: 500;
  opacity: 0.9;
}

.template-2-header-4 {
  font-size: 0.65rem;
  opacity: 0.8;
  font-weight: 700;
}

.template-2-header-5 {
  font-size: 0.8rem;
  opacity: 0.8;
  font-weight: 700;
}

.template-2-header-6 {
  font-size: 0.8rem;
  opacity: 0.9;
  font-weight: 700;
}

@media screen and (max-width: 991px) {
  .main-box-div-template-2 {
    position: relative;
    .img-bg-template-2.col {
      img {
        max-width: 100%;
        height: 590px;
      }
    }
  }
}

@media screen and (max-width: 767px) {
  .main-box-div-template-2 {
    position: relative;
    .img-bg-template-2.col {
      img {
        max-width: 100%;
        height: 710px;
      }
    }
  }
}

@media screen and (max-width: 576px) {
  .main-box-div-template-2 {
    position: relative;
    .img-bg-template-2.col {
      img {
        max-width: 100%;
        height: 850px;
      }
    }
  }
}

// styles for template-03

.template-3-header-0 {
  font-size: 1.5rem;
  font-weight: 500;
}

.template-3-header-1 {
  font-size: 2rem;
  font-weight: 500;
  line-height: 1.2;
}

.template-3-header-2 {
  font-size: 0.85rem;
  opacity: 0.8;
  font-weight: 800;
}

.template-3-header-2-blue {
  font-size: 1.2rem;
  opacity: 0.9;
  font-weight: 500;
  color: #114eaf;
  margin: 0;
}

.template-3-header-3 {
  font-size: 1rem;
  font-weight: 500;
  opacity: 0.9;
}

.template-3-header-4 {
  font-size: 0.75rem;
  opacity: 0.8;
  font-weight: 700;
}

.template-3-header-5 {
  font-size: 0.8rem;
  opacity: 0.9;
  font-weight: 500;
}

.template-3-header-6 {
  font-size: 0.8rem;
  opacity: 0.9;
  font-weight: 700;
}

.template-3-bg-automobile {
  background: #f1f1f1;
  padding: 1rem;
}

.template-3-company-style-1 {
  padding: 10px;
  margin: 35px;
  text-align: center;
}

.template-3-company-style-2 {
  padding: 2px 10px;
  width: auto;
}
.cus-height {
  background-image: url('http://localhost:3000/assets/images/certTempBG03.png');
  background-color: black;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
}
.text-center.bg-automobile.cus-height {
  background: #f0f0f0;
  padding: 2rem;
}

.template-3-footer-left {
  background-color: #265eb5;
  padding: 0.8rem;
}

.template-3-footer-right {
  background-color: #fff;
  padding-left: 1.9rem;
  padding-right: 1rem;
  padding-top: 1rem;
}
