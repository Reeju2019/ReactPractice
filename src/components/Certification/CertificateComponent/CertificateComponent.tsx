import React from "react";
import { Col, Container, FormLabel, Row, Card } from "react-bootstrap";

import PdfGen from "./PdfGen/PdfGen";
import Share from "./Share/Share";

import "./styles.scss";

/**
 * @description Props of Carousal Component
 * @memberof CertificateComponent
 * @property {any} data Config data
 * @property {any} localeConfig Locale config
 * @property {number} selectedTemplate Set the selected template
 */
/* eslint-disable */
interface IProps {
  data: any;
  localeConfig: any;
  selectedTemplate: number;
}

/**
 * @description States of Carousal Component
 * @memberof CertificateComponent
 */
/* eslint-disable */
interface IState {}

/**
 * @component
 * It is child component of Certificate component
 * @description Certificate Component
 */
class CertificateComponent extends React.Component<IProps, IState> {
  state: IState = {};

  pdfRef: any;

  /**
   * @constructor
   * @description Constructor
   * @param {IProps} props Props
   */
  constructor(props: IProps) {
    super(props);
    this.pdfRef = React.createRef();
  }

  render() {
    console.log(this.props.selectedTemplate);
    return (
      <>
        {this.props.selectedTemplate >= 1 && this.props.selectedTemplate <= 3 && (
          <>
            <Row>
              <Col xs={4} md={4}>
                <Row className="header-3">
                  {this.props.localeConfig.certification_header_1}
                </Row>
                <Row className="header-4">
                  {`Template 0${this.props.selectedTemplate}`}
                </Row>
              </Col>
              <Col xs={4} md={7} className="text-end">
                <PdfGen
                  downloadButtonName={
                    this.props.localeConfig.certification_download_button
                  }
                  options={this.props.data.downloadOptions}
                  filename={this.props.data.filename}
                  downloadRef={this.pdfRef}
                />
              </Col>
              <Share
                shareUrl={this.props.data.shareUrl}
                header={
                  this.props.localeConfig.certification_share_modal_header
                }
                closeButtonName={
                  this.props.localeConfig.certification_share_modal_close
                }
                shareButtonName={
                  this.props.localeConfig.certification_share_button
                }
                options={this.props.data.sharingOptions}
              />
            </Row>

            <Row className="mt-5">
              {this.props.selectedTemplate == 1 ? (
                <>
                  <FormLabel
                    ref={this.pdfRef}
                    className="main-box-div-template-1"
                  >
                    <Container>
                      <Row>
                        <Col className="img-bg-template-1">
                          <img src={this.props.data.bgUrl} alt="" />
                        </Col>
                      </Row>
                    </Container>
                    <Container className="content-template-1">
                      <Row className="mb-3 d-flex justify-content-between">
                        <Col className="template-1-header-1" xs={4} md={5}>
                          {this.props.localeConfig.template_header}
                        </Col>
                        <Col
                          xs={4}
                          md={5}
                          className="d-flex flex-column justify-content-end"
                        >
                          <FormLabel className="template-1-header-1 ms-auto">
                            {this.props.data.companyName}
                          </FormLabel>
                          <Card style={{ width: "5rem" }} className="ms-auto">
                            <Card.Img
                              variant="top"
                              src={this.props.data.logoUrl}
                            />
                          </Card>
                        </Col>
                      </Row>
                      <Row className="mb-3">
                        <Col xs={10} md={10}>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.localeConfig.template_1_body_1}
                          </FormLabel>
                          <FormLabel className="header-2-violet ps-1 pe-1">
                            {this.props.data.candidateName}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.localeConfig.template_1_body_2}
                          </FormLabel>
                          <FormLabel className="header-2-violet ps-1 pe-1">
                            {this.props.data.courseName}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.localeConfig.template_1_body_3}
                          </FormLabel>
                          <FormLabel className="template-1-header-1 ps-1 pe-1">
                            {this.props.data.companyName}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.localeConfig.template_1_body_4}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.data.authorisationDate}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.localeConfig.template_1_body_5}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1">
                            {this.props.data.duration}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1">
                            (
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.data.startDate}
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            -
                          </FormLabel>
                          <FormLabel className="template-1-header-3 ps-1 pe-1">
                            {this.props.data.finishDate}
                          </FormLabel>
                          <FormLabel className="template-1-header-3">
                            ).
                          </FormLabel>
                        </Col>
                      </Row>
                      <Row className="mb-3">
                        <Col xs={4} md={8} className="text-center">
                          <Card style={{ width: "5rem" }} className="m-auto">
                            <Card.Img
                              variant="top"
                              src={this.props.data.logoUrl}
                            />
                          </Card>
                        </Col>
                      </Row>
                      <Row className="d-flex justify-content-between">
                        <Col xs={7} md={7}>
                          <FormLabel className="template-1-header-4 ps-1 pe-1">
                            {this.props.localeConfig.template_1_body_6}
                          </FormLabel>
                          <FormLabel className="template-1-header-5 ps-1 pe-1">
                            {this.props.data.website}
                          </FormLabel>
                          <Row>
                            <Col className="">
                              <FormLabel className="template-1-header-4 d-inline ps-1 pe-1">
                                {this.props.data.companyName}
                              </FormLabel>
                              <FormLabel className="template-1-header-5 d-inline ps-1 pe-1">
                                {this.props.localeConfig.template_1_body_7}
                              </FormLabel>
                            </Col>
                          </Row>
                        </Col>
                        <Col xs={5} md={3} className="ps-auto">
                          <Row>signature</Row>
                          <Row className="header-4">
                            {this.props.data.courseManager}
                          </Row>
                          <Row className="header-5">
                            {this.props.localeConfig.template_1_body_8}
                          </Row>
                          <Row className="header-4">
                            {this.props.data.companyName}
                          </Row>
                        </Col>
                      </Row>
                    </Container>
                  </FormLabel>
                </>
              ) : null}

              {this.props.selectedTemplate == 2 ? (
                <>
                  <FormLabel
                    ref={this.pdfRef}
                    className="main-box-div-template-2"
                  >
                    <Container>
                      <Row>
                        <Col className="img-bg-template-2">
                          <img src={this.props.data.bgUrl} alt="" />
                        </Col>
                      </Row>
                    </Container>
                    <Container className="content-template-2">
                      <Row className="mb-3 d-flex justify-content-between">
                        <Col className="template-2-header-1" xs={4} md={4}>
                          {this.props.data.companyName}
                        </Col>
                        <Col xs={4} md={4}>
                          <Card style={{ width: "5rem" }} className="me-3">
                            <Card.Img
                              variant="top"
                              src={this.props.data.logoUrl}
                            />
                          </Card>
                        </Col>
                      </Row>
                      <Row xs={12} md={12} className="mt-5 mb-5 text-center">
                        <Col className="template-2-header-4" xs={12} md={12}>
                          {this.props.localeConfig.template_2_body_1}
                        </Col>
                        <Col
                          className="template-2-header-2-blue"
                          xs={12}
                          md={12}
                        >
                          {this.props.data.candidateName}
                        </Col>
                        <Col className="template-2-header-4" xs={12} md={12}>
                          {this.props.localeConfig.template_2_body_2}
                        </Col>
                        <Col
                          className="template-2-header-2-blue"
                          xs={12}
                          md={12}
                        >
                          {this.props.data.courseName}
                        </Col>
                        <Col className="template-2-header-4" xs={12} md={12}>
                          <FormLabel className="pe-1">
                            {this.props.localeConfig.template_2_body_3}
                          </FormLabel>
                          <FormLabel className="pe-1">
                            {this.props.data.authorisationDate}
                          </FormLabel>
                          <FormLabel className="pe-1">
                            {this.props.localeConfig.template_2_body_4}
                          </FormLabel>
                          <FormLabel className="pe-1">
                            {this.props.data.duration}
                          </FormLabel>
                          <FormLabel className="">(</FormLabel>
                          <FormLabel className="pe-1">
                            {this.props.data.startDate}
                          </FormLabel>
                          <FormLabel className="pe-1">-</FormLabel>
                          <FormLabel className="">
                            {this.props.data.finishDate}
                          </FormLabel>
                          <FormLabel className="">)</FormLabel>
                        </Col>
                        <Col className="template-2-header-4" xs={12} md={12}>
                          <FormLabel className="pe-1">
                            {this.props.localeConfig.template_2_body_5}
                          </FormLabel>
                          <FormLabel>{this.props.data.companyName}</FormLabel>
                        </Col>
                      </Row>
                      <Row className="d-flex justify-content-between">
                        <Col xs={6} md={6}>
                          <Row>
                            <FormLabel className="template-2-header-6 ps-1 pe-1">
                              {this.props.localeConfig.template_2_body_6}
                              <FormLabel className="header-5 ps-1 pe-1">
                                {this.props.data.website}
                              </FormLabel>
                            </FormLabel>
                          </Row>
                          <Row className="d-inline">
                            <FormLabel className="template-2-header-6 ps-1 pe-1">
                              {this.props.data.companyName}
                              <FormLabel className="template-2-header-5 ps-1 pe-1 d-inline">
                                {this.props.localeConfig.template_2_body_7}
                              </FormLabel>
                            </FormLabel>
                          </Row>
                        </Col>
                        <Col xs={6} md={3}>
                          <Row>signature</Row>
                          <Row className="template-2-header-4">
                            {this.props.data.courseManager}
                          </Row>
                          <Row className="template-2-header-5">
                            {this.props.localeConfig.template_2_body_8}
                          </Row>
                          <Row className="template-2-header-4">
                            {this.props.data.companyName}
                          </Row>
                        </Col>
                      </Row>
                    </Container>
                  </FormLabel>
                </>
              ) : null}

              {this.props.selectedTemplate == 3 ? (
                <>
                  <FormLabel ref={this.pdfRef}>
                    <Container className="mt-2">
                      <Row className="">
                        <Row className="d-flex justify-content-between">
                          <Col
                            xs={5}
                            md={5}
                            className="d-flex flex-column justify-content-between cus-height align-items-center p-0"
                          >
                            <Row
                              xs={1}
                              md={2}
                              className="template-3-company-style-1 bg-white ms-2 template-3-header-0"
                            >
                              {this.props.data.companyName}
                            </Row>
                          </Col>
                          <Col
                            className="text-center bg-automobile cus-height"
                            xs={7}
                            md={7}
                          >
                            <Row
                              className="d-flex justify-content-end"
                              xs={12}
                              md={12}
                            >
                              <Card
                                style={{ width: "7rem" }}
                                className="me-3 rounded-circle"
                              >
                                <Card.Img
                                  variant="top"
                                  src={this.props.data.logoUrl}
                                />
                              </Card>
                            </Row>
                            <Row className="text-start">
                              <Col
                                className="template-3-header-4"
                                xs={12}
                                md={12}
                              >
                                {this.props.localeConfig.template_3_body_1}
                              </Col>
                              <Col
                                className="template-3-header-2-blue"
                                xs={12}
                                md={12}
                              >
                                {this.props.data.candidateName}
                              </Col>
                              <Col
                                className="template-3-header-4"
                                xs={12}
                                md={12}
                              >
                                {this.props.localeConfig.template_3_body_2}
                              </Col>
                              <Col
                                className="template-3-header-2-blue"
                                xs={12}
                                md={12}
                              >
                                {this.props.data.courseName}
                              </Col>
                            </Row>
                            <Row className="mt-5 text-start">
                              <Col xs={12} md={12}>
                                signature
                              </Col>
                              <Col
                                xs={12}
                                md={12}
                                className="template-3-header-4"
                              >
                                {this.props.data.courseManager}
                              </Col>
                              <Col
                                xs={12}
                                md={12}
                                className="template-3-header-5"
                              >
                                {this.props.localeConfig.template_3_body_8}
                              </Col>
                              <Col
                                xs={12}
                                md={12}
                                className="template-3-header-2"
                              >
                                {this.props.data.companyName}
                              </Col>
                            </Row>
                          </Col>
                        </Row>
                        <Row className="d-flex">
                          <Col xs={5} md={5}>
                            <Row className="d-flex flex-column justify-content-between template-3-footer-left">
                              <FormLabel className="template-3-company-style-2 text-white template-3-header-5">
                                {this.props.localeConfig.template_3_body_5}
                              </FormLabel>
                              <FormLabel className="template-3-company-style-2 bg-white template-3-header-5">
                                {this.props.data.companyName}
                              </FormLabel>
                            </Row>
                          </Col>
                          <Col xs={7} md={7}>
                            <Row className="d-flex justify-content-between template-3-footer-right">
                              <Col>
                                <Row>
                                  <FormLabel className="template-3-header-6 ps-1 pe-1 m-0">
                                    {this.props.localeConfig.template_2_body_6}
                                    <FormLabel className="header-5 ps-1 pe-1">
                                      {this.props.data.website}
                                    </FormLabel>
                                  </FormLabel>
                                </Row>
                                <Row className="d-inline">
                                  <FormLabel className="template-3-header-6 ps-1 pe-1 m-0">
                                    {this.props.data.companyName}
                                    <FormLabel className="template-3-header-5 ps-1 pe-1 d-inline">
                                      {
                                        this.props.localeConfig
                                          .template_2_body_7
                                      }
                                    </FormLabel>
                                  </FormLabel>
                                </Row>
                              </Col>
                            </Row>
                          </Col>
                        </Row>
                      </Row>
                    </Container>
                  </FormLabel>
                </>
              ) : null}
            </Row>
          </>
        )}
      </>
    );
  }
}

export { CertificateComponent };
export default CertificateComponent;
