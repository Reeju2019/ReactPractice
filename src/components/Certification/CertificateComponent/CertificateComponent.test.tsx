import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import CertificateComponent from "./CertificateComponent"

const data = {
  id: 1,
  companyName: "Company Name",
  logoUrl: "/assets/images/company-logo.png",
  bgUrl: "/assets/images/certTempBG01.png",
  candidateName: "Ketan Kuperkar",
  courseName: "Automobile Basics Course - Level II",
  duration: "6 months",
  authorisationDate: "20th August",
  startDate: "17 March",
  finishDate: "19 August",
  website: "demo.company.website.com",
  courseManager: "HEMA PARULKAR",
  shareUrl: "https://www.w3schools.com/",
  filename: "demo.cert01.pdf",
  sharingOptions: ["facebook", "twitter"],
  downloadOptions: {
    orientation: "p",
    unit: "mm",
    format: "a4",
    putOnlyUsedFonts: true,
  },
}

const localeConfig = {
  uploadAsset_header_1: "Upload your",
  uploadAsset_header_2: "Asset",
  uploadAsset_subHeader_1: "Supported File Formats:",
  uploadAsset_subHeader_2: "img, jpeg, png and etc",
  uploadAsset_dropBoxHeader_1: "Drop your files here or",
  uploadAsset_dropBoxHeader_2: "Browse",
  uploadAsset_dropBoxSubHeader_1: "Max size of Video: 25 MB",
  uploadAsset_upload_msg_success: "uploading Complete!",
  uploadAsset_upload_msg_failure:
    "something went wrong, please check file type/size and upload again!",
  reportTable_header: "Courses",
  reportTable_viewDetails: "View Details",
  reportTable_viewAll: "View All",
  columnChart_header: "Grade Reports",
  doughnutChart_header: "Grade Reports",
  doughnutChart_legendHeader: "LEGEND",
  certification_header_1: "Certificate Preview",
  certification_download_button: "Download Certificate",
  certification_share_button: "Share",
  certification_share_modal_header: "Share Your Certificate",
  certification_share_modal_close: "Close",
  template_header: "COURSE CERTIFICATE",
  template_1_body_1: "This is to certify that",
  template_1_body_2: "has successfully completed the",
  template_1_body_3: "an online course authorized by",
  template_1_body_4: "on",
  template_1_body_5: "for the course of",
  template_1_body_6: "VERIFY AT",
  template_1_body_7:
    "has confirmed the identity of this individual and their participation in the course.",
  template_1_body_8: "Course Manager",
  template_2_body_1: "THIS IS TO CERTIFY THAT",
  template_2_body_2: "HAS SUCCESSFULLY COMPLETED",
  template_2_body_3: "ON",
  template_2_body_4: "FOR THE COURSE OF",
  template_2_body_5: "AUTHORIZED BY",
  template_2_body_6: "VERIFY AT",
  template_2_body_7:
    "has confirmed the identity of this individual and their participation in the course.",
  template_2_body_8: "Course Manager",
  template_3_body_1: "This is to certify that",
  template_3_body_2: "has successfully completed",
  template_3_body_4: "FOR THE COURSE OF",
  template_3_body_5: "This course is authorized by",
  template_3_body_6: "VERIFY AT",
  template_3_body_7:
    "has confirmed the identity of this individual and their participation in the course.",
  template_3_body_8: "Course Manager",
}

/**
 * Unit testing for Certificate Component
 */
describe("CertificateComponent Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <CertificateComponent
        data={data}
        localeConfig={localeConfig}
        selectedTemplate={1}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <CertificateComponent
        data={data}
        localeConfig={localeConfig}
        selectedTemplate={1}
      />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
