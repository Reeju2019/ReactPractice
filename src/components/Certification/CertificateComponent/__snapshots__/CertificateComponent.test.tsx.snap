// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificateComponent Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <React.Fragment>
      <Row>
        <Col
          md={4}
          xs={4}
        >
          <Row
            className="header-3"
          >
            Certificate Preview
          </Row>
          <Row
            className="header-4"
          >
            Template 01
          </Row>
        </Col>
        <Col
          className="text-end"
          md={7}
          xs={4}
        >
          <PdfGen
            downloadButtonName="Download Certificate"
            downloadRef={
              Object {
                "current": null,
              }
            }
            filename="demo.cert01.pdf"
            options={
              Object {
                "format": "a4",
                "orientation": "p",
                "putOnlyUsedFonts": true,
                "unit": "mm",
              }
            }
          />
        </Col>
        <Share
          closeButtonName="Close"
          header="Share Your Certificate"
          options={
            Array [
              "facebook",
              "twitter",
            ]
          }
          shareButtonName="Share"
          shareUrl="https://www.w3schools.com/"
        />
      </Row>
      <Row
        className="mt-5"
      >
        <React.Fragment>
          <FormLabel
            className="main-box-div-template-1"
            column={false}
            visuallyHidden={false}
          >
            <Container
              fluid={false}
            >
              <Row>
                <Col
                  className="img-bg-template-1"
                >
                  <img
                    alt=""
                    src="/assets/images/certTempBG01.png"
                  />
                </Col>
              </Row>
            </Container>
            <Container
              className="content-template-1"
              fluid={false}
            >
              <Row
                className="mb-3 d-flex justify-content-between"
              >
                <Col
                  className="template-1-header-1"
                  md={5}
                  xs={4}
                >
                  COURSE CERTIFICATE
                </Col>
                <Col
                  className="d-flex flex-column justify-content-end"
                  md={5}
                  xs={4}
                >
                  <FormLabel
                    className="template-1-header-1 ms-auto"
                    column={false}
                    visuallyHidden={false}
                  >
                    Company Name
                  </FormLabel>
                  <Card
                    body={false}
                    className="ms-auto"
                    style={
                      Object {
                        "width": "5rem",
                      }
                    }
                  >
                    <CardImg
                      src="/assets/images/company-logo.png"
                      variant="top"
                    />
                  </Card>
                </Col>
              </Row>
              <Row
                className="mb-3"
              >
                <Col
                  md={10}
                  xs={10}
                >
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    This is to certify that
                  </FormLabel>
                  <FormLabel
                    className="header-2-violet ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    Ketan Kuperkar
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    has successfully completed the
                  </FormLabel>
                  <FormLabel
                    className="header-2-violet ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    Automobile Basics Course - Level II
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    an online course authorized by
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-1 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    Company Name
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    on
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    20th August
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    for the course of
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    6 months
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    (
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    17 March
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    -
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    19 August
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-3"
                    column={false}
                    visuallyHidden={false}
                  >
                    ).
                  </FormLabel>
                </Col>
              </Row>
              <Row
                className="mb-3"
              >
                <Col
                  className="text-center"
                  md={8}
                  xs={4}
                >
                  <Card
                    body={false}
                    className="m-auto"
                    style={
                      Object {
                        "width": "5rem",
                      }
                    }
                  >
                    <CardImg
                      src="/assets/images/company-logo.png"
                      variant="top"
                    />
                  </Card>
                </Col>
              </Row>
              <Row
                className="d-flex justify-content-between"
              >
                <Col
                  md={7}
                  xs={7}
                >
                  <FormLabel
                    className="template-1-header-4 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    VERIFY AT
                  </FormLabel>
                  <FormLabel
                    className="template-1-header-5 ps-1 pe-1"
                    column={false}
                    visuallyHidden={false}
                  >
                    demo.company.website.com
                  </FormLabel>
                  <Row>
                    <Col
                      className=""
                    >
                      <FormLabel
                        className="template-1-header-4 d-inline ps-1 pe-1"
                        column={false}
                        visuallyHidden={false}
                      >
                        Company Name
                      </FormLabel>
                      <FormLabel
                        className="template-1-header-5 d-inline ps-1 pe-1"
                        column={false}
                        visuallyHidden={false}
                      >
                        has confirmed the identity of this individual and their participation in the course.
                      </FormLabel>
                    </Col>
                  </Row>
                </Col>
                <Col
                  className="ps-auto"
                  md={3}
                  xs={5}
                >
                  <Row>
                    signature
                  </Row>
                  <Row
                    className="header-4"
                  >
                    HEMA PARULKAR
                  </Row>
                  <Row
                    className="header-5"
                  >
                    Course Manager
                  </Row>
                  <Row
                    className="header-4"
                  >
                    Company Name
                  </Row>
                </Col>
              </Row>
            </Container>
          </FormLabel>
        </React.Fragment>
      </Row>
    </React.Fragment>
  </React.Fragment>,
]
`;
